### PR TITLE
PauseMenuDataMgr: use better variable names

### PR DESCRIPTION
**Pull request by Kinak338**
I haven't made any changes before so please tell me if I make any mistakes

- - -
This change is [Reviewable](https://reviewable.io/reviews/zeldaret/botw/123)

### CODE REVIEW

File: src/Game/UI/uiPauseMenuDataMgr.h [hunk 1]
```diff
@@ -236,7 +236,7 @@ class PouchItem {
     ItemUse mItemUse = ItemUse::Invalid;
     s32 mValue = 0;
     bool mEquipped = false;
-    u8 _25 = 1;
+    u8 mInInventory = 1;
```
Comment (leoetlino) [inline src/Game/UI/uiPauseMenuDataMgr.h:239]: Seems like this should be a boolean (bool) instead of a u8.
```suggestion
    bool mInInventory = 1;
```

File: src/Game/UI/uiPauseMenuDataMgr.h [hunk 2]
```diff
@@ -50,7 +50,7 @@ static_assert(NumPouchItemsMax == 420, "NumPouchItemsMax must be 420 for now");
 constexpr int ItemStackSizeMax = 999;
 
 // TODO: figure out what this is
-constexpr int NumPouch50 = 50;
+constexpr int NumTabMax = 50;
```
Comment (leoetlino) [inline src/Game/UI/uiPauseMenuDataMgr.h:53]: can you clarify what this is?

File: src/Game/UI/uiPauseMenuDataMgr.h [hunk 3]
```diff
@@ -174,7 +174,7 @@ class PouchItem {
 
     PouchItemType getType() const { return mType; }
     bool isEquipped() const { return mEquipped; }
-    u8 get25() const { return _25; }
+    u8 isInInventory() const { return mInInventory; }
```
Comment (leoetlino) [inline src/Game/UI/uiPauseMenuDataMgr.h:177]:
```suggestion
    bool isInInventory() const { return mInInventory; }
```

File: src/Game/UI/uiPauseMenuDataMgr.h [hunk 4]
```diff
@@ -479,12 +479,12 @@ class PauseMenuDataMgr {
...
     sead::SafeArray<PouchItem**, NumPouchCategories> mListHeads;
-    sead::SafeArray<PouchItem*, NumPouch50> mArray1;
-    sead::SafeArray<PouchItemType, NumPouch50> mArray2;
+    sead::SafeArray<PouchItem*, NumTabMax> mArrayTabs;
+    sead::SafeArray<PouchItemType, NumTabMax> mArrayTabsType;
     PouchItem* mLastAddedItem{};
-    s32 _44490 = -1;
-    s32 _44494 = -1;
-    s32 _44498{};
+    u32 mLastAddedItemTab = 0xFFFFFFFF;
```
Comment (leoetlino) [inline src/Game/UI/uiPauseMenuDataMgr.h:485]: why was this changed to an unsigned int?

File: src/Game/UI/uiPauseMenuDataMgr.h [hunk 5]
```diff
@@ -479,12 +479,12 @@ class PauseMenuDataMgr {
...
+    s32 mLastAddedItemSlot = -1;
+    s32 mNumTab = 0;
```
Comment (leoetlino) [inline src/Game/UI/uiPauseMenuDataMgr.h:487]:
```suggestion
    s32 mNumTabs = 0;
```

File: src/Game/UI/uiPauseMenuDataMgr.h [hunk 6]
```diff
@@ -499,7 +499,7 @@ class PauseMenuDataMgr {
     PouchItem* mGoronSoulItem{};
     PouchItem* mZoraSoulItem{};
     PouchItem* mGerudoSoulItem{};
-    bool _44538 = false;
+    bool mCanSeeHealthBar = false;
```
Comment (leoetlino) [inline src/Game/UI/uiPauseMenuDataMgr.h:502]: can you add a comment to clarify what "health bar" this is referring to?

**Comment by leoetlino** [comment]
also please fix the clang-format failure -- either run clang-format to reformat the code automatically, or reformat it manually by doing what it tells you to do.

File: src/Game/UI/uiPauseMenuDataMgr.h (hunk 4 above)
Comment (Kinak338) [reply src/Game/UI/uiPauseMenuDataMgr.h:485]: I have seen this line in updateInventoryInfo : " *(uint *)(param_1 + 0x44490) = uVar20 - 1; "
I made this change because uVar20 is the number of tab you have in your inv

Comment (leoetlino) [reply src/Game/UI/uiPauseMenuDataMgr.h:485]: That line alone does not tell you if it's signed or unsigned - Ghidra is only making a guess (especially when you haven't provided it with the proper type for `param_1`) and chances are that the assembly corresponding to that single line is exactly the same whether mLastAddedItemTab is signed or unsigned.

IMO it's more likely for this to be signed, given the -1 value, the other signed ints right next to this variable and Nintendo EPD tending to use signed ints.

File: src/Game/UI/uiPauseMenuDataMgr.h [hunk 7]
```diff
@@ -49,8 +49,8 @@ static_assert(NumPouchItemsMax == 420, "NumPouchItemsMax must be 420 for now");
 
 constexpr int ItemStackSizeMax = 999;
 
-// TODO: figure out what this is
-constexpr int NumPouch50 = 50;
+// maximum number of tab (pages with 20 items) you can have, more will be glitched
```
Comment (leoetlino) [inline src/Game/UI/uiPauseMenuDataMgr.h:52]:
```suggestion
// Maximum number of tabs (pages with 20 items) you can have.
// Going beyond this limit will glitch the menu.
```

File: src/Game/UI/uiPauseMenuDataMgr.h [hunk 8]
```diff
@@ -499,7 +499,9 @@ class PauseMenuDataMgr {
     PouchItem* mGoronSoulItem{};
     PouchItem* mZoraSoulItem{};
     PouchItem* mGerudoSoulItem{};
-    bool _44538 = false;
+
+    // Indicates if you have the champion tunic or a divine helmet eqquiped and so if you can see enemy hp
```
Comment (leoetlino) [inline src/Game/UI/uiPauseMenuDataMgr.h:503]: Thanks for the explanation. Minor nits:

```suggestion
    // Indicates if the Champion's Tunic or a divine helm is equipped
    // and whether the player can see enemy HP.
```

**Review by leoetlino — APPROVED** [review]
Thanks, this looks good! Just one last thing: Could you rename mArrayTabs to mTabs and mArrayTabsType to mTabTypes? I put "array" in the placeholder name because otherwise it wasn't obvious that those variables were arrays, but now that we know those are tabs/tab types we don't need to keep "array" in the names.
